sys/registry/int_path.c

```diff
@@ -0,0 +1,426 @@
+            return 0;
+        }
+    }
+
+    for (size_t i = 0; i < schema->groups_len; i++) {
+        registry_parameter_t *found_parameter = (registry_parameter_t *)_internal_parameter_lookup(
+            schema->groups[i],
+            parameter_id);
+        if (found_parameter != NULL) {
+            *parameter = found_parameter;
+            return 0;
+        }
+    }
+
+    return -REGISTRY_ERROR_PARAMETER_NOT_FOUND;
+}
+
+/* to int path */
+registry_namespace_int_path_t registry_to_namespace_int_path(const registry_namespace_t *namespace)
+{
+    assert(namespace != NULL);
+
+    return (registry_namespace_int_path_t) {
+               .namespace_id = namespace->id,
+    };
+}
+
+registry_schema_int_path_t registry_to_schema_int_path(const registry_schema_t *schema)
+{
+    assert(schema != NULL);
+
+    return (registry_schema_int_path_t) {
+               .namespace_id = schema->namespace->id,
+               .schema_id = schema->id,
+    };
+}
+
+registry_instance_int_path_t registry_to_instance_int_path(const registry_instance_t *instance)
+{
+    assert(instance != NULL);
+
+    return (registry_instance_int_path_t) {
+               .namespace_id = instance->schema->namespace->id,
+               .schema_id = instance->schema->id,
+               .instance_id = instance->id,
+    };
+}
+
+registry_group_int_path_t registry_to_group_int_path(const registry_instance_t *instance,
+                                                     const registry_group_t *group)
+{
+    assert(instance != NULL);
+    assert(group != NULL);
+
+    return (registry_group_int_path_t) {
+               .namespace_id = instance->schema->namespace->id,
+               .schema_id = instance->schema->id,
+               .instance_id = instance->id,
+               .group_id = group->id,
+    };
+}
+
+registry_parameter_int_path_t registry_to_parameter_int_path(const registry_instance_t *instance,
+                                                             const registry_parameter_t *parameter)
+{
+    assert(instance != NULL);
+    assert(parameter != NULL);
+
+    return (registry_parameter_int_path_t) {
+               .namespace_id = instance->schema->namespace->id,
+               .schema_id = instance->schema->id,
+               .instance_id = instance->id,
+               .parameter_id = parameter->id,
+    };
+}
+
+/* from int path */
+int registry_from_namespace_int_path(const registry_namespace_int_path_t *path,
+                                     registry_namespace_t **namespace)
+{
+    assert(path != NULL);
+    assert(namespace != NULL);
+
+    int res = _namespace_lookup(path->namespace_id, namespace);
+
+    return res;
+}
+
+int registry_from_schema_int_path(const registry_schema_int_path_t *path,
+                                  registry_namespace_t **namespace, registry_schema_t **schema)
+{
+    assert(path != NULL);
+    /* leaving namespace NULL is fine */
```

Comment:
It is fine, because `namespace` is an output parameter.
If the user of the API is only interested in the `schema`, then the user can just put `NULL` to the namespace as a way of telling the API: I am not interested in the namespace, just give me the schema.

As with the other issues before. Once I changed this to have only one function to rule them all, we will automatically fix this as well. (Because there will only be one `registry_node_t` or similar name? output struct. instead of having `namespace` and `registry` separately.

Reply:
This is outdated and fixed. This function does not exist anymore